### ISSUE TITLE
Suggestion: Additive design verse subtractive.

### ISSUE THREAD

**Issue by Fallen-Entertainment** [suriyun-production/mmorpg-kit-docs#1128]
Currently I find myself stripping down the Demo often. It occurred to me that because of partials / inherit this kit is kinda perfect for a more modular work flow. If features were all done like modules / addons users could simply drop it what they are interested in. This is more of a set up / presentation thing which shouldnt take to much work. I feel like it would make the kit alot easier to work with and also provide a base demo thats stripped down already ready to go. It seemed like such a good idea that I figured I would suggest it to you. Thanks for your time and thank you for making the best kit of its kind out there! <3

**Comment by shubhank008** [comment]
I also agree on this, the kit is super powerful and modular enough to customize it for many kinds of games, but it feels just too congested right now that gives a hard time for any entry/new buyer and even setting up any kind of new project initially.  
For starters, I think the demo scenes included in the kit should be packaged and included as .unitypackage so there are not so many folders and files and scenes, anyone needing a particular demo can extract it.  

Secondly, I think the components need more decluterring specially into more grouped ones, character/player entity component is quite damn large, so is gameinstance.  

And lastly, I also agree with Fallen's part that the kit needs to be broken down to modules which you can enable or add when needed, a lot of the kit's systems might not be needed for starting projects or different use cases.  

The kit is so awesome (just like your shooter/battle kits were) that I find using them for any kind of multiplayer game needs I might have, even if not mmorpg. So modularity will just increase use case and possibilities to do things with this kit, at a faster pace.  
Right now just as Fallen said, have to strip a lot of unneeded code or components and maintain git repos of different versions to use.

**Comment by gordowebdesign** [comment]
+1

**Comment by insthync** [comment]
I think it's gone too far to make changes, do you guys have any suggestion for changing?

**Comment by shubhank008** [comment]
> I think it's gone too far to make changes, do you guys have any suggestion for changing?

My personal suggestion is to break down the Playable component to two or more, I find thats the one single biggest and complex component.  
Perhaps separate the equipment/model containers and animations to their own components ?

Otherwise, I also think a lot of things are too entangled now to separate them, but perhaps advanced RPG stuff like crafting/recipes, harvesting, guilds, etc. can be turned into addons or plug-in systems.  
Similarly the 2D and demos can be turned into extractable-unity packages.  

I think for most of us, having a minimal initial version of the kit and then keep adding/enabling features that are needed would make it easier to use the kit(code), lightweighted and easier to manage.  

Personally, I always start with a very minimal version of the kit (feature wise), create my MVP/Demo and then start enabling more advanced features like guild, party, advanced gamerules/calculations, etc.

---Edit---
But yes, to be honest, now that I am more accustomed to using the kit, I agree that the codebase of kit is too far to untangle anything. But perhaps if some of the components can be dissected, that alone would be a huge improvement and faster workflow.

**Comment by GeoffCee** [comment]
My thought on this is moving towards having a core kit package with seperate unity packages for RPG, Survival etc would be good. So initially only the core asset is installed with a folder containing the demo packages for the specific genres. The demo scene files and entities for a specific game genre could be installed using these other Unity packages.

**Comment by insthync** [comment]
How to make changes without "remake it all" feeling :P

**Comment by shubhank008** [comment]
> How to make changes without "remake it all" feeling :P

The part about excluding various demos from the initial install of the kit can be done through unity packages like many other assets do, as then people who donot want the demos can delete the .unitypackage files while those who want can simply extract them.  

Rest all is how you see (if) can be done and whether its priority enough ? I don't think so.  
Personally I would be content with just breaking down the playable component to two-three components itself, my personal gripe is only with how huge that one component is, that if you need to change animations and expand it, it goes off my screen vertically, and I have a huge screen.

I had previously attempted separating animations and equipment/container parts of it to separate components, but failed miserably.  

Again, none of them are urgent enough to need your time, I see you are working on addressables, which I feel would be more more important to help improve performance and memory usage of the kit, and potentially implement a Patch/Update system in future.
